### PR TITLE
feature: handle in memory binary streams

### PR DESCRIPTION
**Pull request by sylvainlb**

### CODE REVIEW

**Comment by MKoesters** [comment]
Hi and thanks for contributing to pymzML!

I already had a look at your code and it looks fine.
As soon as I have fixed the appveyor encoding issue and verified everything is running as expected, I'll merge your branch into master.

Best,
Manuel

**Comment by sylvainlb** [comment]
Thanks a lot Manuel!

How often do you push updates on pypi?

**Comment by MKoesters** [comment]
Rather infrequently, unfortunately.
However we're working on automating uploading new releases and I hope we can get this up and running as soon as possible.
@JB-MS, can you upload this as soon as I have fixed the encoding Issue?

**Comment by sylvainlb** [comment]
Thanks!

Keep me posted.

File: pymzml/run.py [hunk 1]
```diff
@@ -220,14 +229,7 @@ def _determine_file_encoding(self, path):
         mzml_encoding = 'utf-8'
```
Comment (MKoesters) [inline pymzml/run.py]: Please change to:

```
if os.path.exists(path):
    if path.endswith('.gz') or path.endswith('.igz'):
        import gzip
        _open = gzip.open
    else:
        _open = open
    with open(path, 'rb') as sniffer:
        header = sniffer.readline()
```
So we can also guess the encoding in gzip compressed files

**Comment by MKoesters** [comment]
Hi,
I've added a comment for changing the _determine_file_encoding function.

Additional to this, can you please change test_determine_file_encoding function in main_reader_test.py to
`
encoding = self.reader_compressed_indexed._determine_file_encoding(self.paths[2])
self.assertEqual(encoding, 'ISO-8859-1')
encoding = self.reader_compressed_unindexed._determine_file_encoding(self.paths[1])
self.assertEqual(encoding, 'ISO-8859-1')
encoding = self.reader_uncompressed_indexed._determine_file_encoding(self.paths[3])
self.assertEqual(encoding, 'ISO-8859-1')
encoding = self.reader_uncompressed_unindexed._determine_file_encoding(self.paths[0])
self.assertEqual(encoding, 'ISO-8859-1')
`

The test were faulty and it took me some time to get a running windows machine.
However I tested the change and tests were succeeding.
Since I dont know how to change a pull request and did not want to make a new one, I hope you can add another commit so we can merge your branch.
If you have another solution, please let  me know.

Best,
Manuel

**Comment by MKoesters** [comment]
Hi @sylvainlb ,

I fixed your pull request and opened a new one.
I'll merge this after talking to the rest of the team and make sure the new release gets uploaded to pypi.
I'll give you notice here, as soon as the new package is uploaded

**Comment by sylvainlb** [comment]
Thanks!

Do you require some action from me?

**Pull request closed without merging**